frontend/src/lib/k8s/cluster.ts

```diff
@@ -429,6 +443,38 @@ export function makeKubeObject<T extends KubeObjectInterface | KubeEvent>(
     static get isNamespaced() {
       return this.apiEndpoint.isNamespaced;
     }
+    static get readOnlyFields(): JsonPath<T>[] {
+      return this._readOnlyFields;
+    }
+    getEditableObject() {
+      const fieldsToRemove = this._class().readOnlyFields;
+      const code = this.jsonData ? cloneDeep(this.jsonData) : '{}';
```

Comment:
The type of code will be different depending on whether we have jsonData or not (i.e. the else part should be {}, not '{}').